File: mvpreaderjava/src/main/java/com/example/mvpreaderjava/di/module/AppModule.java
package com.example.mvpreaderjava.di.module;

import com.example.mvpreaderjava.MyApplication;
import com.example.mvpreaderjava.modle.DataManager;
import com.example.mvpreaderjava.modle.helper.HttpHelper;
import com.example.mvpreaderjava.modle.helper.HttpManager;
import com.google.gson.Gson;

import java.security.PublicKey;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by sw116 on 2018/4/19.
 */
@Module
public class AppModule {

    private MyApplication application;

    public AppModule(MyApplication application) {
        this.application = application;
    }

    @Provides
    @Singleton
    public MyApplication provideApp() {
        return application;
    }

    @Provides
    @Singleton
    public Gson provideGson() {
        return new Gson();
    }

    @Provides
    @Singleton
    public HttpHelper provideHttpHelper(HttpManager manager) {
        return manager;
    }

    @Provides
    @Singleton
    public DataManager provideDataManager(HttpHelper manager) {
        return new DataManager(manager);
    }
}

File: app/src/main/java/com/savion/mvpreader/di/module/AppModule.kt
package com.savion.mvpreader.di.module

import com.savion.mvpreader.MvpApplication
import com.savion.mvpreader.model.DataManager
import com.savion.mvpreader.model.behavior.HttpHelper
import com.savion.mvpreader.model.http.HttpManager
import dagger.Module
import dagger.Provides
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by savion on 2018/4/10.
 */
@Module
class AppModule @Inject constructor(val application: MvpApplication) {

    @Provides
    @Singleton
    fun provideApp(): MvpApplication {
        return application
    }

    @Provides
    @Singleton
    fun provideHttpManager(helper: HttpManager): HttpHelper {
        return helper
    }
    @Provides
    @Singleton
    fun provideDataManager(dataManager: DataManager): DataManager {
        return dataManager
    }

}